### ISSUE TITLE
I need help about "Error: The requested platform 'ZWayServer' has been registered multiple times."

### ISSUE THREAD

**Issue by senadaruc** [SphtKr/homebridge-zway#22]
Dears,

I installed the homebridge-zway plugin and configured my config file like this:

```
{
    "bridge": {
        "name": "Homebridge",
        "username": "CC:22:3D:E3:CE:30",
        "port": 51826,
        "pin": "031-45-154"
    },
    "platforms": [
        {
            "platform": "Nest",
            "token": "c.E8NL5PnWVbTRc4fRxxxxM4Fw6ACdaoNoOEju4oRK4RAPgVg4z4irFnLKrqKS3JNI6GNwFqq9CGsPYbMRVKkeItVu3xxceG1zz",
            "clientId": "f74548xxxdbd128879",
            "clientSecret": "rvTv4I1xxxx91ofju",
            "code": "Wxxx33F",
            "username": "xxxx@gmail.com",
            "password": "xxxxxx"
        },
        {
            "platform": "ZWayServer",
            "url": "http://10.0.0.3:8083/",
            "login": "admin",
            "password": "xxxxxxx"
        }
    ]
}
```

The error I receive after running the homebridge

```
pi@pi-sony-down:~ $ homebridge 
*** WARNING *** The program 'node' uses the Apple Bonjour compatibility layer of Avahi.
*** WARNING *** Please fix your application to use the native API of Avahi!
*** WARNING *** For more information see <http://0pointer.de/avahi-compat?s=libdns_sd&e=node>
*** WARNING *** The program 'node' called 'DNSServiceRegister()' which is not supported (or only supported partially) in the Apple Bonjour compatibility layer of Avahi.
*** WARNING *** Please fix your application to use the native API of Avahi!
*** WARNING *** For more information see <http://0pointer.de/avahi-compat?s=libdns_sd&e=node&f=DNSServiceRegister>
[3/26/2016, 11:27:34 AM] Loaded plugin: homebridge-legacy-plugins
[3/26/2016, 11:27:34 AM] Registering accessory 'homebridge-legacy-plugins.AD2USB'
[3/26/2016, 11:27:34 AM] Registering accessory 'homebridge-legacy-plugins.Carwings'
[3/26/2016, 11:27:34 AM] Registering accessory 'homebridge-legacy-plugins.ELKM1'
[3/26/2016, 11:27:34 AM] Registering accessory 'homebridge-legacy-plugins.FileSensor'
[3/26/2016, 11:27:34 AM] Registering accessory 'homebridge-legacy-plugins.GenericRS232Device'
[3/26/2016, 11:27:34 AM] Registering accessory 'homebridge-legacy-plugins.HomeMatic'
[3/26/2016, 11:27:34 AM] Registering accessory 'homebridge-legacy-plugins.HomeMaticThermo'
[3/26/2016, 11:27:34 AM] Registering accessory 'homebridge-legacy-plugins.HomeMaticWindow'
[3/26/2016, 11:27:34 AM] Registering accessory 'homebridge-legacy-plugins.HttpGarageDoorOpener'
[3/26/2016, 11:27:34 AM] Registering accessory 'homebridge-legacy-plugins.HttpHygrometer'
[3/26/2016, 11:27:34 AM] Registering accessory 'homebridge-legacy-plugins.HttpThermometer'
[3/26/2016, 11:27:34 AM] Registering accessory 'homebridge-legacy-plugins.Tesla'
[3/26/2016, 11:27:34 AM] Registering accessory 'homebridge-legacy-plugins.X10'
[3/26/2016, 11:27:34 AM] Registering accessory 'homebridge-legacy-plugins.mpdclient'
[3/26/2016, 11:27:34 AM] Registering platform 'homebridge-legacy-plugins.Domoticz'
[3/26/2016, 11:27:34 AM] Registering platform 'homebridge-legacy-plugins.HomeAssistant'
[3/26/2016, 11:27:34 AM] Registering platform 'homebridge-legacy-plugins.HomeSeer'
[3/26/2016, 11:27:34 AM] Registering platform 'homebridge-legacy-plugins.ISY'
[3/26/2016, 11:27:34 AM] Registering platform 'homebridge-legacy-plugins.LIFx'
[3/26/2016, 11:27:34 AM] Registering platform 'homebridge-legacy-plugins.SmartThings'
[3/26/2016, 11:27:34 AM] Registering platform 'homebridge-legacy-plugins.TelldusLive'
[3/26/2016, 11:27:34 AM] Registering platform 'homebridge-legacy-plugins.ZWayServer'
[3/26/2016, 11:27:34 AM] ---
[3/26/2016, 11:27:36 AM] Loaded plugin: homebridge-nest
[3/26/2016, 11:27:36 AM] Registering platform 'homebridge-nest.Nest'
[3/26/2016, 11:27:36 AM] ---
[3/26/2016, 11:27:36 AM] Loaded plugin: homebridge-zway
[3/26/2016, 11:27:36 AM] Registering accessory 'homebridge-zway.ZWayServer'
[3/26/2016, 11:27:36 AM] Registering platform 'homebridge-zway.ZWayServer'
[3/26/2016, 11:27:36 AM] ---
[3/26/2016, 11:27:36 AM] Loaded config.json with 0 accessories and 2 platforms.
[3/26/2016, 11:27:36 AM] ---
[3/26/2016, 11:27:36 AM] Loading 2 platforms...
[3/26/2016, 11:27:36 AM] Initializing Nest platform...
[3/26/2016, 11:27:36 AM] Fetching Nest devices.
/usr/local/lib/node_modules/homebridge/lib/api.js:103
      throw new Error("The requested platform '" + name + "' has been registered multiple times. Please be more specific by writing one of: " + found.join(", "));
      ^

Error: The requested platform 'ZWayServer' has been registered multiple times. Please be more specific by writing one of: homebridge-legacy-plugins.ZWayServer, homebridge-zway.ZWayServer
    at API.platform (/usr/local/lib/node_modules/homebridge/lib/api.js:103:13)
    at Server._loadPlatforms (/usr/local/lib/node_modules/homebridge/lib/server.js:263:45)
    at Server.run (/usr/local/lib/node_modules/homebridge/lib/server.js:71:36)
    at module.exports (/usr/local/lib/node_modules/homebridge/lib/cli.js:40:10)
    at Object.<anonymous> (/usr/local/lib/node_modules/homebridge/bin/homebridge:17:22)
    at Module._compile (module.js:434:26)
    at Object.Module._extensions..js (module.js:452:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Function.Module.runMain (module.js:475:10)
    at startup (node.js:117:18)
    at node.js:951:3
pi@pi-sony-down:~ $ 
```
What I am doing wrong?

Thanks,

**Comment by SphtKr** [comment]
Hi @senadaruc , First off, try `npm upgrade -g homebridge-legacy-plugins`. The latest version of that shouldn't have that old version of ZWayServer included, so if you upgrade it it should remove the extra copy.

Second, you may wish to `npm upgrade -g homebridge` also. I _think_ the latest versions of Homebridge deal better with multiple copies of the plugin by ignoring the second one it encounters--or at least that's what I see happen when I'm developing.

**Comment by senadaruc** [comment]
Thanks for the prompt response so the legacy plugins contain all the plugins and I don't need to install them separately?

Thanks.

**Comment by SphtKr** [comment]
No, the legacy plugins are older ones that were built in a time when all the available plugins were part of the Homebridge project itself. Then the API changed and there were too many to manage in one project. I updated the ZWayServer plugin to the new API and broke it out into a separate project (like most other plugins), so it has been removed from the legacy package. The old plugins in the legacy project will probably continue to work, but if Homebridge changes again they may break.
